packages/nimble-components/src/wafer-map/modules/wafer-map-converter.ts

```diff
@@ -0,0 +1,45 @@
+import { Field, List, Table, Utf8, vectorFromArray } from 'apache-arrow';
+import type { WaferMapDie, WaferMapLayerData } from '../types';
+
+/**
+ * This class is used to convert old wafer map data to new wafer map data
+ */
+export class WaferMapConverter {
```

Comment:
So, again, I think the lack of context for how this class will be used is problematic. It's hard to understand the utility of this functionality even being encapsulated in a separate class. For instance, if there was a single place where we intended for either of these "conversions" to occur, then why can we not just make these private methods on the `WaferMap` component class?

Further, is there a use-case in mind for when `populateLayers` would be called outside of the context of the `toApacheTable` method? If not, then why is it public? These are the kinds of questions that are more easily answered when this implementation is accompanied with its usage in production.

Ultimately, while I think smaller PRs are a good goal, there are limits before they become problematic. At worst, you would be adding 4 files to a different PR, two of which are extremely light. My preference would be to bundle this PR with another.

Reply:
I discussed with @munteannatan and you are right, we will postpone the work on this task until the infrastructure will allow this API to be actually used. If we will continue on this pr or we will reuse the code in a different one will be decided later.